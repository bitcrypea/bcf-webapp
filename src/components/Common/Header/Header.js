import React, { Component, Fragment } from 'react';
import { Menu, Dropdown, Icon } from 'antd';
// import { Link } from 'react-router-dom';
import { push } from 'connected-react-router';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {
  HeaderContainerHeader,
  HeaderTopNavHead,
  HeaderTopNavMain,
  HeaderTopNavLeft,
  HeaderTopNavRight,
  HeaderTagA,
  HeaderMenuTitle,
  HeaderLogo,
  HeaderLoginButton,
  HeaderRegister,
} from './styled';
import logo from '../../../assets/images/logo.png';
import { isLoggedIn } from '../../../redux/auth/selectors';
import { logout } from '../../../redux/auth/actions';

class Header extends Component {
  handleLogout = () => {
    this.props.logout();
    this.props.gotoHome();
  };

  render() {
    const { authenticated } = this.props;
    return (
      <HeaderContainerHeader>
        <HeaderTopNavHead>
          <HeaderTopNavMain>
            <HeaderTopNavLeft>
              <HeaderTagA onClick={() => this.props.gotoHome()}>
                <HeaderLogo src={logo} width="90" height="60" />
              </HeaderTagA>
              {/* <Link style={{ color: 'white' }} to="/exchange">
                Exchange
              </Link> */}
            </HeaderTopNavLeft>
            <HeaderTopNavRight>
              {authenticated && (
                <Dropdown
                  overlay={
                    <Menu selectedKeys={[]}>
                      <Menu.Item
                        style={{ paddingLeft: 18, paddingRight: 18 }}
                        key="userinfo"
                        onClick={this.props.gotoMyAccount}
                      >
                        <Icon type="setting" />
                        My Account
                      </Menu.Item>
                      <Menu.Divider />
                      <Menu.Item
                        style={{ paddingLeft: 18, paddingRight: 18 }}
                        onClick={this.handleLogout}
                        key="logout"
                      >
                        <Icon type="logout" />
                        Logout
                      </Menu.Item>
                    </Menu>
                  }
                  placement="bottomRight"
                  trigger={['click']}
                >
                  <HeaderMenuTitle>
                    <Icon
                      style={{ fontSize: 24, color: 'white' }}
                      type="user"
                      theme="outlined"
                    />
                  </HeaderMenuTitle>
                </Dropdown>
              )}
              {!authenticated && (
                <Fragment>
                  <HeaderLoginButton
                    style={{ color: 'white' }}
                    onClick={() => this.props.gotoLogin()}
                  >
                    Login
                  </HeaderLoginButton>
                  <HeaderRegister onClick={() => this.props.gotoRegister()}>
                    Sign up
                  </HeaderRegister>
                </Fragment>
              )}
            </HeaderTopNavRight>
          </HeaderTopNavMain>
        </HeaderTopNavHead>
      </HeaderContainerHeader>
    );
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      logout,
      gotoMyAccount: () => push('/my-account'),
      gotoRegister: () => push('/register'),
      gotoLogin: () => push('/login'),
      gotoHome: () => push('/'),
    },
    dispatch
  );

const mapStateToProps = state => ({
  authenticated: isLoggedIn(state),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
