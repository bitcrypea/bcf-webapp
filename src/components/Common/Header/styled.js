import styled from 'styled-components';

export const HeaderMenuTitle = styled.div`
  color: #666;
  cursor: pointer;
`;

export const HeaderLogo = styled.img`
  cursor: pointer;
  border-style: none;
`;

export const HeaderLoginButton = styled.a`
  cursor: pointer;
  text-decoration: none;
  color: #659aea;
  background-color: transparent;
`;

export const HeaderRegister = styled.a`
  cursor: pointer;
  color: #fff;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  height: 32px;
  line-height: 32px;
  border-radius: 3px;
  margin: 0 0 0 16px;
  background-color: #659aea;
`;

export const HeaderTagA = styled.a`
  margin-right: 32px;
  background-color: transparent;
  color: rgb(102, 102, 102);
`;

export const HeaderContainerHeader = styled.div`
  padding: 0px;
  width: 100%;
  background: #fff;
  height: 64px;
  line-height: 64px;
  flex: 0 0 auto;
`;

export const HeaderTopNavHead = styled.div`
  width: 100%;
  background: rgb(44, 40, 39);
  transition: background 0.3s, width 0.2s;
  height: 64px;
  padding: 0 12px 0 0;
  box-shadow: 0 1px 4px rgba(0, 21, 41, 0.08);
  position: relative;
`;

export const HeaderTopNavMain = styled.div`
  max-width: 1200px;
  margin: auto;
  padding-left: 4px;
  display: flex;
  height: 64px;
  justify-content: space-between;
`;

export const HeaderTopNavLeft = styled.div`
  color: white;
  display: flex;
`;
export const HeaderTopNavRight = styled.div``;
