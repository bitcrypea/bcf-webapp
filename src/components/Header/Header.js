import React, { Component } from "react";
import { Menu, Dropdown, Button } from "antd";
import { Link } from "react-router-dom";

import {
  NewNav,
  Container,
  NavLeft,
  NavRight,
  TagA,
  Logo,
  LoginButton,
  Register,
  MenuTitle
} from "./styled";
import logo from "../../logo.png";

const menuTokenTrading = (
  <Menu>
    <Menu.Item>
      <Link to="/exchange">Exchange</Link>
    </Menu.Item>
    <Menu.Item>
      <a
        target="_blank"
        rel="noopener noreferrer"
        href="#"
      >
        Full Screen
      </a>
    </Menu.Item>
  </Menu>
);


class Header extends Component {
  render() {
    return (
      <Container>
        <NewNav>
          <NavLeft>
            <TagA href="/">
              <Logo src={logo} width="104" height="32" />
            </TagA>
            <Dropdown overlay={menuTokenTrading} placement="bottomLeft">
              <MenuTitle>Token Trading</MenuTitle>
            </Dropdown>
          </NavLeft>
          <NavRight>
            <LoginButton>Login</LoginButton>
            <Register>Sign up</Register>
          </NavRight>
        </NewNav>
      </Container>
    );
  }
}

export default Header;
