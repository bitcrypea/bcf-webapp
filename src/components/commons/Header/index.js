import React, { Component, Fragment } from 'react';
import { Menu, Dropdown, Icon } from 'antd';
// import { Link } from 'react-router-dom';
import { push } from 'connected-react-router';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import {
  HeaderContainerHeader,
  HeaderTopNavHead,
  HeaderTopNavMain,
  HeaderTopNavLeft,
  HeaderTopNavRight,
  HeaderTagA,
  HeaderMenuTitle,
  HeaderLogo,
  HeaderLoginButton,
  HeaderRegister
} from './styled';
import logo from '../../../assets/images/logo.png';
import { isLoggedIn } from '../../../redux/auth/selectors';
import { logout } from '../../../redux/auth/actions';
import { persistor } from '../../../redux/store';
import { createNewAddress } from '../../../redux/pusher/actions';

class Header extends Component {
  handleLogout = () => {
    const { createNewAddress } = this.props;
    localStorage.removeItem('TOKEN_ID');
    localStorage.removeItem('TOKEN_SECRET');
    localStorage.removeItem('affiliate_codes');
    createNewAddress('');
    persistor.purge();

    this.props.logout();
    this.props.gotoHome();
  };

  render() {
    const {
      authenticated,
      gotoHome,
      gotoMyAccount,
      gotoLogin,
      gotoRegister
    } = this.props;
    return (
      <HeaderContainerHeader>
        <HeaderTopNavHead>
          <HeaderTopNavMain>
            <HeaderTopNavLeft>
              <HeaderTagA onClick={() => gotoHome()}>
                <HeaderLogo src={logo} width="90" height="60" />
              </HeaderTagA>
              {/* <Link style={{ color: 'white' }} to="/exchange">
                Exchange
              </Link> */}
            </HeaderTopNavLeft>
            <HeaderTopNavRight>
              {authenticated && (
                <Dropdown
                  overlay={
                    <Menu selectedKeys={[]}>
                      <Menu.Item
                        style={{ paddingLeft: 18, paddingRight: 18 }}
                        key="userinfo"
                        onClick={gotoMyAccount}
                      >
                        <Icon type="setting" />
                        My Account
                      </Menu.Item>
                      <Menu.Divider />
                      <Menu.Item
                        style={{ paddingLeft: 18, paddingRight: 18 }}
                        onClick={this.handleLogout}
                        key="logout"
                      >
                        <Icon type="logout" />
                        Logout
                      </Menu.Item>
                    </Menu>
                  }
                  placement="bottomRight"
                  trigger={['click']}
                >
                  <HeaderMenuTitle>
                    <Icon
                      style={{ fontSize: 24 }}
                      type="user"
                      theme="outlined"
                    />
                  </HeaderMenuTitle>
                </Dropdown>
              )}
              {!authenticated && (
                <Fragment>
                  <HeaderLoginButton onClick={() => gotoLogin()}>
                    Login
                  </HeaderLoginButton>
                  <HeaderRegister onClick={() => gotoRegister()}>
                    Sign up
                  </HeaderRegister>
                </Fragment>
              )}
            </HeaderTopNavRight>
          </HeaderTopNavMain>
        </HeaderTopNavHead>
      </HeaderContainerHeader>
    );
  }
}

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      createNewAddress,
      logout,
      gotoMyAccount: () => push('/my-account'),
      gotoRegister: () => push('/register'),
      gotoLogin: () => push('/login'),
      gotoHome: () => push('/')
    },
    dispatch
  );

const mapStateToProps = state => ({
  authenticated: isLoggedIn(state)
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
