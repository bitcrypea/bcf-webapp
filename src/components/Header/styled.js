import styled from 'styled-components'

export const MenuTitle = styled.div`
  color: #666;
  cursor: pointer;
`

export const Container = styled.header`
  background: #fff;
  min-width: 1200px;
  padding: 0 30px;
  font-family: Open Sans, SF Pro Text, Helvetica Neue, Helvetica, Arial,
    sans-serif;
  font-size: 16px;
  line-height: 1.5;
`

export const NewNav = styled.div`
  display: flex;
  justify-content: space-between;
  height: 60px;
  background-color: #fff;
  font-family: Open Sans, SF Pro Text, Helvetica Neue, Helvetica, Arial,
    sans-serif;
`

export const NavLeft = styled.div`
  max-width: 800px;
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: flex;
  align-items: center;
  font-size: 14px;
`

export const NavRight = styled.div`
  max-width: 800px;
  list-style-type: none;
  padding: 0;
  margin: 0;
  display: flex;
  align-items: center;
  font-size: 12px;
`

export const TagA = styled.a`
  margin-right: 32px;
  background-color: transparent;
`

export const Logo = styled.img`
  cursor: pointer;
  border-style: none;
`

export const LoginButton = styled.a`
  cursor: pointer;
  text-decoration: none;
  color: #659aea;
  background-color: transparent;
`

export const Register = styled.a`
  cursor: pointer;
  color: #fff;
  padding-left: 16px;
  padding-right: 16px;
  height: 32px;
  line-height: 32px;
  border-radius: 3px;
  margin: 0 0 0 16px;
  background-color: #659aea;
`
