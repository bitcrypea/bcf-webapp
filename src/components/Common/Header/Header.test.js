import React from 'react';
import renderer from 'react-test-renderer';
import Header from './Header';

it('Header renders correctly', () => {
  const tree = renderer.create(<Header fdfadsf />).toJSON();
  expect(tree).toMatchSnapshot();
});
