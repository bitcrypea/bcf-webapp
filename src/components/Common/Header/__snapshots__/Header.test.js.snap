// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="sc-bwzfXH cAOBC"
>
  <div
    className="sc-htpNat gLXFtu"
  >
    <div
      className="sc-bxivhb hUIpRx"
    >
      <a
        className="sc-EHOje kdUijg"
        href="/"
      >
        <img
          className="sc-bZQynM kQBLMx"
          height="40"
          src="logo-bitcrypea.png"
          width="104"
        />
      </a>
      <div
        className="ant-dropdown-trigger sc-bdVaJa kdWzoU"
        disabled={undefined}
        onBlur={undefined}
        onClick={undefined}
        onContextMenu={undefined}
        onFocus={undefined}
        onMouseDown={undefined}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={undefined}
      >
        Token Trading
      </div>
    </div>
    <div
      className="sc-ifAKCX diTFmv"
    >
      <a
        className="sc-gzVnrw cTTjsC"
        href="/login"
      >
        Login
      </a>
      <a
        className="sc-htoDjs eOHVTT"
        href="/register"
      >
        Sign up
      </a>
    </div>
  </div>
</header>
`;
